### ISSUE TITLE
Latest 2.5-alpine image incompatible with pg gem.

### ISSUE THREAD

**Issue by cma-arnold** [docker-library/ruby#227]
The latest 2.5-alpine image appears to be incompatible with the pg gem (version 1.1.x) leaving rails apps that use a postgres database unable to run.

Steps to reproduce:
```
$ docker images | grep alpine
ruby                                                                    2.5-alpine          d822253438ab        5 days ago          62MB
alpine                                                                  latest              11cd0b38bc3c        7 weeks ago         4.41MB
$ docker run -it --rm ruby:2.5-alpine sh
/ # apk --no-cache add build-base ruby-dev postgresql-dev
fetch http://dl-cdn.alpinelinux.org/alpine/v3.7/main/x86_64/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/v3.7/community/x86_64/APKINDEX.tar.gz
(1/27) Installing binutils-libs (2.30-r1)
(2/27) Installing binutils (2.30-r1)
(3/27) Installing gmp (6.1.2-r1)
(4/27) Installing isl (0.18-r0)
(5/27) Installing libgomp (6.4.0-r5)
(6/27) Installing libatomic (6.4.0-r5)
(7/27) Installing libgcc (6.4.0-r5)
(8/27) Installing mpfr3 (3.1.5-r1)
(9/27) Installing mpc1 (1.0.3-r1)
(10/27) Installing libstdc++ (6.4.0-r5)
(11/27) Installing gcc (6.4.0-r5)
(12/27) Installing musl-dev (1.1.18-r3)
(13/27) Installing libc-dev (0.7.1-r0)
(14/27) Installing g++ (6.4.0-r5)
(15/27) Installing make (4.2.1-r0)
(16/27) Installing fortify-headers (0.9-r0)
(17/27) Installing build-base (0.5-r0)
(18/27) Installing db (5.3.28-r0)
(19/27) Installing libsasl (2.1.26-r11)
(20/27) Installing libldap (2.4.45-r3)
(21/27) Installing libpq (10.5-r0)
(22/27) Installing postgresql-libs (10.5-r0)
(23/27) Installing postgresql-dev (10.5-r0)
(24/27) Installing libgmpxx (6.1.2-r1)
(25/27) Installing gmp-dev (6.1.2-r1)
(26/27) Installing ruby-libs (2.4.4-r0)
(27/27) Installing ruby-dev (2.4.4-r0)
Executing busybox-1.27.2-r11.trigger
OK: 211 MiB in 58 packages
/ # gem install -v 1.1.0 pg
Fetching: pg-1.1.0.gem (100%)
Building native extensions. This could take a while...
Successfully installed pg-1.1.0
1 gem installed
/ # irb
irb(main):001:0> require 'pg'
Traceback (most recent call last):
        8: from /usr/local/bin/irb:11:in `<main>'
        7: from (irb):1
        6: from /usr/local/lib/ruby/site_ruby/2.5.0/rubygems/core_ext/kernel_require.rb:39:in `require'
        5: from /usr/local/lib/ruby/site_ruby/2.5.0/rubygems/core_ext/kernel_require.rb:135:in `rescue in require'
        4: from /usr/local/lib/ruby/site_ruby/2.5.0/rubygems/core_ext/kernel_require.rb:135:in `require'
        3: from /usr/local/bundle/gems/pg-1.1.0/lib/pg.rb:4:in `<top (required)>'
        2: from /usr/local/lib/ruby/site_ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
        1: from /usr/local/lib/ruby/site_ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
LoadError (Error relocating /usr/local/bundle/gems/pg-1.1.0/lib/pg_ext.so: rb_enc_alias: symbol not found - /usr/local/bundle/gems/pg-1.1.0/lib/pg_ext.so)
```

**Comment by aiomaster** [comment]
pg gem 1.0.0 works.
See these issues:

- https://bitbucket.org/ged/ruby-pg/issues/281/pg-gem-110-compiles-on-alpine-linux-but
- https://github.com/ged/ruby-pg/pull/28

**Comment by yosifkit** [comment]
This was fixed in pg gem `1.1.2`: https://bitbucket.org/ged/ruby-pg/pull-requests/28

On a side note, do not install the `ruby-dev` package. It is not related to the ruby installed in the image. The ruby in the image is installed from source and may not be the same version or even compatible with `ruby-dev` from alpine packages.
```console
$ docker run -it --rm ruby:2.5-alpine sh
/ # apk --no-cache add build-base postgresql-dev
fetch http://dl-cdn.alpinelinux.org/alpine/v3.7/main/x86_64/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/v3.7/community/x86_64/APKINDEX.tar.gz
(1/23) Installing binutils-libs (2.30-r1)
(2/23) Installing binutils (2.30-r1)
(3/23) Installing gmp (6.1.2-r1)
(4/23) Installing isl (0.18-r0)
(5/23) Installing libgomp (6.4.0-r5)
(6/23) Installing libatomic (6.4.0-r5)
(7/23) Installing libgcc (6.4.0-r5)
(8/23) Installing mpfr3 (3.1.5-r1)
(9/23) Installing mpc1 (1.0.3-r1)
(10/23) Installing libstdc++ (6.4.0-r5)
(11/23) Installing gcc (6.4.0-r5)
(12/23) Installing musl-dev (1.1.18-r3)
(13/23) Installing libc-dev (0.7.1-r0)
(14/23) Installing g++ (6.4.0-r5)
(15/23) Installing make (4.2.1-r0)
(16/23) Installing fortify-headers (0.9-r0)
(17/23) Installing build-base (0.5-r0)
(18/23) Installing db (5.3.28-r0)
(19/23) Installing libsasl (2.1.26-r11)
(20/23) Installing libldap (2.4.45-r3)
(21/23) Installing libpq (10.5-r0)
(22/23) Installing postgresql-libs (10.5-r0)
(23/23) Installing postgresql-dev (10.5-r0)
Executing busybox-1.27.2-r11.trigger
OK: 197 MiB in 54 packages
/ # gem install -v 1.1.2 pg
Fetching: pg-1.1.2.gem (100%)
Building native extensions. This could take a while...
Successfully installed pg-1.1.2
1 gem installed
/ # irb
irb(main):001:0> require 'pg'
=> true
irb(main):002:0> 
/ # 
```

Closing since the docker images are fine and it is a problem with a specific version of the pg gem.
